lib/ansible/modules/cloud/vmware/vmware_vsan_health_info.py

```diff
@@ -0,0 +1,177 @@
+
+author:
+    - Erwan Quelin (@equelin)
+'''
+
+EXAMPLES = '''
+- name: Gather health info from a vSAN's cluster
+  hosts: localhost
+  gather_facts: 'no'
+  vmware_vsan_health_info:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    cluster_name: 'vSAN01'
+    fetch_from_cache: False
+'''
+
+RETURN = '''
+vsan_health_info:
+    description: vSAN cluster health info
+    returned: on success
+    type: dict
+    sample: {
+        "_vimtype": "vim.cluster.VsanClusterHealthSummary",
+        "burnInTest": null,
+        "clusterStatus": {
+            "_vimtype": "vim.cluster.VsanClusterHealthSystemStatusResult",
+            "goalState": "installed",
+            "status": "green",
+            "trackedHostsStatus": [
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi01.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi04.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi02.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi03.example.com",
+                    "issues": [],
+                    "status": "green"
+                }
+            ],
+            "untrackedHosts": []
+        }
+    }
+'''
+
+import json
+import traceback
+
+try:
+    from pyVmomi import vim, vmodl, VmomiSupport
+    HAS_PYVMOMI = True
+    HAS_PYVMOMIJSON = hasattr(VmomiSupport, 'VmomiJSONEncoder')
+except ImportError:
+    PYVMOMI_IMP_ERR = traceback.format_exc()
+    HAS_PYVMOMI = False
+    HAS_PYVMOMIJSON = False
+
+try:
+    import vsanapiutils
+    import vsanmgmtObjects
+    HAS_VSANPYTHONSDK = True
+except ImportError:
+    VSANPYTHONSDK_IMP_ERR = traceback.format_exc()
+    HAS_VSANPYTHONSDK = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.vmware import connect_to_api, vmware_argument_spec, find_cluster_by_name
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        cluster_name=dict(required=True, type='str'),
+        fetch_from_cache=dict(required=False, type='bool')
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_PYVMOMI:
+        module.fail_json(msg=missing_required_lib('PyVmomi'), exception=PYVMOMI_IMP_ERR)
+
+    if not HAS_VSANPYTHONSDK:
```

Comment:
Do you have a little doc that explain how to install the lib? I've just downloaded `vsan-sdk-python.zip` from VMware website, but I don't see any installation procedure.

This dependency sounds rather problematic to me because it's not part of the existing requirements of the other VMware modules, in addition, it's not a freeesoftware. Do you know any alternative?

Reply:
You will find documentation in the README file (at the root of the zip file) and also in this blog post - https://www.virtuallyghetto.com/2016/03/getting-started-with-the-new-vsan-6-2-management-api.html

As far as I know, this SDK is the only available and also the only supported by VMware. I asked some guys at VMware if it will be open sourced one day but I didn't get any answers...

Is it really an issue as far as this requirement is stated in the documentation and checks are made in the code ?